### PR TITLE
Round15: lr-warmup-epochs=2 — extend warmup sweep (SOTA=1, confirmed winner)

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

1-epoch linear LR warmup (PR #222, fern) confirmed a **−1.51% test improvement** (10.420% vs prior SOTA 10.580%). The warmup helps by letting the optimizer build up gradient-direction confidence before committing to full LR=1e-4. The question is whether 2 epochs of warmup (slower ramp) is better than 1, or whether it cuts too much into the productive flat-LR training phase.

At 1-epoch warmup with T_max=50 (effectively flat LR), the effective schedule is:
- Ep1: LR ramps from ~0 → 1e-4 (warmup)
- Ep2–9: LR ≈ 1e-4 (flat, minimal cosine decay)

At 2-epoch warmup:
- Ep1–2: LR ramps from ~0 → 1e-4
- Ep3–9: LR ≈ 1e-4 (flat)

One fewer epoch of productive gradient descent, but gentler entry into full LR. This is a single-delta test to close the warmup sweep.

**Single delta vs SOTA:** `--lr-warmup-epochs 2` (SOTA uses `--lr-warmup-epochs 1`)

## Instructions

Run the following command **exactly as written** (single delta: only `--lr-warmup-epochs 2` changes):

```bash
cd target/
torchrun --standalone --nproc_per_node=8 train.py \
  --agent chihiro --wandb-name "chihiro/round15-lr-warmup-2ep" \
  --wandb-group "tay-round15-lr-warmup-sweep" \
  --optimizer lion --lr 1e-4 --weight-decay 5e-4 \
  --no-compile-model --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 --model-slices 128 \
  --ema-decay 0.999 --lr-warmup-epochs 2
```

Once launched, post your W&B run IDs (rank0 + all 8 ranks) as a comment on this PR.

Report `val_primary/abupt_axis_mean_rel_l2_pct` at each epoch as it completes.

When training finishes, report final `test_primary/*` metrics.

## Baseline

Current SOTA: **PR #222 fern** (`ut1qmc3i`) — `lr_warmup_epochs=1`

| Epoch | val_abupt |
|-------|-----------|
| ep7   | 9.8759%   |
| ep8   | 9.4516%   |
| **ep9** | **9.2910%** |

| Metric | PR #222 (current SOTA) | AB-UPT |
|--------|------------------------|--------|
| `test_abupt_axis_mean_rel_l2_pct` | **10.420%** | — |
| `test_surface_pressure_rel_l2_pct` | **5.550%** | 3.82% |
| `test_wall_shear_rel_l2_pct` | **10.185%** | 7.29% |
| `test_volume_pressure_rel_l2_pct` | **12.737%** | 6.08% |
| `test_wall_shear_x_rel_l2_pct` | **8.629%** | 5.35% |
| `test_wall_shear_y_rel_l2_pct` | **12.329%** | 3.65% |
| `test_wall_shear_z_rel_l2_pct` | **12.854%** | 3.63% |

**Merge bar:** `val_primary/abupt_axis_mean_rel_l2_pct` < **9.291%** (PR #222 best val).

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: Status check — this PR has been assigned for 24+ hours with no activity comment. Please confirm:

1. Have you launched the training run? Post your W&B run ID (rank0) as a reply to this comment.
2. If you hit any issues (OOM, import error, flag not recognized), describe them immediately.

Your experiment: `--lr-warmup-epochs 2` (single delta vs SOTA). The run command is in the PR body. Launch it now if you haven't already.

**Comment by morganmcg1** [comment]
ADVISOR: Second escalation. You were assigned this experiment over 24 hours ago and no W&B activity has been detected. SOTA is 9.065% and we need these experiments running immediately. If you are having environment or pod issues, reply with the error and I will help unblock you. Otherwise, start training NOW and post your W&B run ID in this PR within the next 2 hours. GPU time is precious — if there is no response within 2 hours, this PR will be reassigned.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #263 because ADVISOR: Closing — this PR was assigned to a student without a DrivAerML pod (this cluster only has alphonse/askeladd/edward/fern/frieren/nezuko/tanjiro/thorfinn). The hypothesis will be re-issued to an actual DrivAerML student in a fresh PR..
